logstash-core/lib/logstash/compiler.rb

```diff
@@ -41,11 +42,23 @@ def self.compile_imperative(source_with_metadata, support_escapes)
       raise ConfigurationError, grammar.failure_reason
     end
 
+    plugins = config.recursive_select(LogStashCompilerLSCLGrammar::LogStash::Compiler::LSCL::AST::Plugin)
+    added_source_line_column(pipeline_config, plugins)
+
     config.process_escape_sequences = support_escapes
     config.compile(source_with_metadata)
   end
 
-  def self.compile_graph(source_with_metadata, support_escapes)
-    Hash[compile_imperative(source_with_metadata, support_escapes).map {|section,icompiled| [section, icompiled.toGraph]}]
+  def self.compile_graph(pipeline_config, source_with_metadata, support_escapes)
+    Hash[compile_imperative(pipeline_config, source_with_metadata, support_escapes).map {|section,icompiled| [section, icompiled.toGraph]}]
   end
+
+  def self.added_source_line_column(pipeline_config, plugins)
+    for plugin in plugins do
```

Comment:
nitpick: use `Enumerable#each`
```suggestion
    plugins.each do |plugin|
```